pgex/examples/collision_shapes/main.py

```diff
@@ -0,0 +1,335 @@
+
+                # Early out if the two reference to same shape
+                if shape_a is shape_b: continue
+
+                # Circle and circle collision
+                if isinstance(shape_a, Circle) and isinstance(shape_b, Circle):
+                    if circle_x_circle(shape_a, shape_b):
+                        collided.append(shape_a)
+                        collided.append(shape_b)
+
+                # Rect and circle collision
+                elif isinstance(shape_a, pygame.FRect) and isinstance(shape_b, Circle):
+                    if rect_x_circle(shape_a, shape_b):
+                        collided.append(shape_a)
+                        collided.append(shape_b)
+
+                # Rect and rect collision
+                elif isinstance(shape_a, pygame.FRect) and isinstance(shape_b, pygame.FRect):
+                    if shape_a.colliderect(shape_b):
+                        collided.append(shape_a)
+                        collided.append(shape_b)
+
+                # Polygon and circle collision
+                elif isinstance(shape_a, Polygon) and isinstance(shape_b, Circle):
+                    if polygon_x_circle(shape_a, shape_b):
+                        collided.append(shape_a)
+                        collided.append(shape_b)
+
+                # Polygon and rect collision
+                elif isinstance(shape_a, Polygon) and isinstance(shape_b, pygame.FRect):
+                    if polygon_x_rect(shape_a, shape_b):
+                        collided.append(shape_a)
+                        collided.append(shape_b)
+
+                # Polygon and polygon collision
+                elif isinstance(shape_a, Polygon) and isinstance(shape_b, Polygon):
+                    if polygon_x_polygon(shape_a, shape_b):
+                        collided.append(shape_a)
+                        collided.append(shape_b)
+
+        # Render scene
+
+        window.fill((255, 255, 255))
+
+        for shape in all_shapes:
+            if shape in collided:
+                color = (187, 255, 77)
+                outline_color = (23, 145, 97)
+            else:
+                color = (255, 144, 18)
+                outline_color = (156, 23, 23)
+
+            if isinstance(shape, Circle):
+                pygame.draw.circle(window, color, shape.center, shape.radius)
+                pygame.draw.circle(window, outline_color, shape.center, shape.radius, 1)
+
+            elif isinstance(shape, pygame.FRect):
+                pygame.draw.rect(window, color, shape)
+                pygame.draw.rect(window, outline_color, shape, 1)
+            
+            elif isinstance(shape, Polygon):
+                pygame.draw.polygon(window, color, shape.local_to_world())
+                pygame.draw.polygon(window, outline_color, shape.local_to_world(), 1)
+
+        pygame.display.flip()
+
+        asyncio.sleep(0)
+
+
+def run() -> None:
+    asyncio.run(main())
+
+
+if __name__ == "__main__":
+    run()
```

Comment:
Missing blank line at the end of the file.
https://www.flake8rules.com/rules/W391.html